pkg/snapshot/restorer/restorer.go

```diff
@@ -123,19 +124,29 @@ func (r *Restorer) restoreFromBaseSnapshot(ro RestoreOptions) error {
 
 	walDir := filepath.Join(memberDir, "wal")
 	snapdir := filepath.Join(memberDir, "snap")
-	if err = makeDB(r.zapLogger, snapdir, ro.BaseSnapshot, len(cl.Members()), r.store, false); err != nil {
+	if err = r.makeDB(r.zapLogger, snapdir, ro.BaseSnapshot, len(cl.Members()), r.store, false); err != nil {
 		return err
 	}
-	return makeWALAndSnap(r.zapLogger, walDir, snapdir, cl, ro.Config.Name)
+	return r.makeWALAndSnap(r.zapLogger, walDir, snapdir, cl, ro.Config.Name)
 }
 
 // makeDB copies the database snapshot to the snapshot directory.
-func makeDB(logger *zap.Logger, snapdir string, snap snapstore.Snapshot, commit int, ss snapstore.SnapStore, skipHashCheck bool) error {
-	rc, err := ss.Fetch(snap)
+func (r *Restorer) makeDB(logger *zap.Logger, snapdir string, snap snapstore.Snapshot, commit int, ss snapstore.SnapStore, skipHashCheck bool) error {
+	reader, err := ss.Fetch(snap)
+
 	if err != nil {
 		return err
 	}
-	defer rc.Close()
+
+	isCompressed := isSnapshotCompressed(snap)
+	if isCompressed {
+		// decompress the snapshots
+		startDeCompressTime := time.Now()
+		reader = compressor.DecompressSnapshots(reader, snap.CompressionSuffix)
+		totalDeCompressTime := time.Now().Sub(startDeCompressTime).Seconds()
+		r.logger.Infof("Decompress the snapshot in %v seconds", totalDeCompressTime)
```

Comment:
This logging should happen after the `io.Copy()` call because the stream reading happens there. -> https://github.com/gardener/etcd-backup-restore/pull/293/files?file-filters%5B%5D=.go#diff-9547d8cc0c155be97c8e3a057b9351b6bf83108543f957fb8bac1adf111ccfaaR160